src/Lachain.Core/Blockchain/VM/ExternalHandler.cs

```diff
@@ -1024,6 +1056,12 @@ public static void Handle_Env_WriteEvent(int signatureOffset, int valueOffset, i
                             throw new InvalidOperationException();
             var value = SafeCopyFromMemory(frame.Memory, valueOffset, valueLength) ??
                         throw new InvalidOperationException();
+            
+            List<UInt256>? topics = null;
+            if (HardforkHeights.IsHardfork_4Active(frame.InvocationContext.Snapshot.Blocks.GetTotalBlockHeight()))
```

Comment:
do we need to create an empty topic list ? As far as I understand, null and empty both represent no topic.

Reply:
Done.